### ISSUE TITLE
[a11y] Menu - Back button is not accessible

### ISSUE THREAD

**Issue by g123k** [openfoodfacts/openfoodfacts-server#10288]
Hi everyone,

To improve accessibility on the website, we have some "simple" fixes to implement.
On the menu (visible in mobile mode), the back button is not labelled correctly:

<img width="352" alt="Screenshot 2024-05-13 at 16 46 40" src="https://github.com/openfoodfacts/openfoodfacts-server/assets/246838/cf0b536b-3270-4f60-8af8-d748ebb7c5c2">
